test_rmw_implementation/test/test_graph_api.cpp

```diff
@@ -0,0 +1,808 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
```

Comment:
Are you planning on adding versions of these tests that check the good cases?

Reply:
Ideally yes, but likely not with this PR. Rationale here is to cover basic error handling cases as described in https://github.com/ros2/rmw/pull/272 only, which are the ones definitely not covered by tests in upper layers.

Reply:
Perfectly reasonable, just wanted to check.